evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -608,3 +607,19 @@ def test_simple_imputer_errors_with_bool_and_categorical_columns(
     else:
         si = SimpleImputer()
         si.fit(X_df)
+
+
+def test_simple_imputer_boolean_nullable_all_null():
+    X_train = pd.DataFrame({"a": [pd.NA] * 20 + [1.0] + [pd.NA] * 20})
+    y = pd.Series(range(len(X_train)))
+    X_test = pd.DataFrame({"a": [pd.NA] * 10})
```

Comment:
Times like these, I think it's helpful to docstring the test to get at what exactly you're testing here.  The test name doesn't seem to match what's going on.  The test case here is that you're train is sparse and your test set happens to not be fully representative of all the classes available in `X`, right?

Reply:
Basically, yeah. It's really just testing having an all-null test set when the training had non-null values. I'll update the test name and add a docstring